From 77f35812b576cdb2f56c6fc4c7661b275aca7495 Mon Sep 17 00:00:00 2001
From: Xu Jia <xujia39@huawei.com>
Date: Sat, 25 Jun 2022 11:46:23 +0800
Subject: [PATCH 2624/4585] [Huawei] ipv6: fix kabi for mc_forwarding in struct
 ipv6_devconf

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187072

--------------------------------

Making mc_forwarding atomic breaks the KABI of struct ipv6_devconf.
This patch uses KABI_REPLACE to fix it.

Fixes: f314decf62ec "([Backport] ipv6: make mc_forwarding atomic)"
Signed-off-by: Xu Jia <xujia39@huawei.com>
---
 include/linux/ipv6.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/ipv6.h b/include/linux/ipv6.h
index f4e1790a5eed..210402b26a20 100644
--- a/include/linux/ipv6.h
+++ b/include/linux/ipv6.h
@@ -51,7 +51,7 @@ struct ipv6_devconf {
 	__s32		use_optimistic;
 #endif
 #ifdef CONFIG_IPV6_MROUTE
-	atomic_t	mc_forwarding;
+	KABI_REPLACE(__s32 mc_forwarding, atomic_t mc_forwarding)
 #endif
 	__s32		disable_ipv6;
 	__s32		drop_unicast_in_l2_multicast;
-- 
2.17.1

